napari/_qt/qt_viewer.py

```diff
@@ -579,47 +538,7 @@ def _add_layer(self, layer):
             if vispy_layer.events is not None:
                 vispy_layer.events.loaded.connect(self._qt_poll.wake_up)
 
-        vispy_layer.node.parent = self.view.scene
-        self.layer_to_visual[layer] = vispy_layer
-
-        # ensure correct canvas blending
-        layer.events.visible.connect(self._reorder_layers)
-
-        self._reorder_layers()
-
-    def _remove_layer(self, event):
-        """When a layer is removed, remove its parent.
-
-        Parameters
-        ----------
-        event : napari.utils.event.Event
-            The napari event that triggered this method.
-        """
-        layer = event.value
-        layer.events.visible.disconnect(self._reorder_layers)
-        vispy_layer = self.layer_to_visual[layer]
-        vispy_layer.close()
-        del vispy_layer
-        del self.layer_to_visual[layer]
-        self._reorder_layers()
-
-    def _reorder_layers(self):
-        """When the list is reordered, propagate changes to draw order."""
-        first_visible_found = False
-        for i, layer in enumerate(self.viewer.layers):
-            vispy_layer = self.layer_to_visual[layer]
-            vispy_layer.order = i
-
-            # the bottommost visible layer needs special treatment for blending
-            if layer.visible and not first_visible_found:
-                vispy_layer.first_visible = True
-                first_visible_found = True
-            else:
-                vispy_layer.first_visible = False
-            vispy_layer._on_blending_change()
-
-        self.canvas._draw_order.clear()
-        self.canvas.update()
+        self.canvas.add_layer_to_visual(layer, vispy_layer)
```

Comment:
It feels like this whole function should happen inside the canvas (i.e: not create the `VispyLayer` at this level). Is there a way to get the "qt_poll" stuff in there?